pkg/apis/cincinnati/v1alpha1/cincinnati_types.go

```diff
@@ -0,0 +1,79 @@
+package v1alpha1
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CincinnatiSpec defines the desired state of Cincinnati
+type CincinnatiSpec struct {
+	// +kubebuilder:validation:Minimum=1
+	// Replicas is the number of pods to run. When >=2, a PodDisruptionBudget
+	// will ensure that voluntary disruption leaves at least one Pod running at
+	// all times.
+	Replicas int32 `json:"replicas"`
+
+	// Registry is the container registry to use, such as "quay.io".
+	Registry string `json:"registry"`
+
+	// Repository is the repository to use in the Registry, such as
+	// "openshift-release-dev/ocp-release"
+	Repository string `json:"repository"`
+
+	// GitHubOrg is the organization to use on GitHub for retrieving additional
+	// graph data.
+	GitHubOrg string `json:"gitHubOrg"`
+
+	// GitHubRepo is the repository to use on GitHub for retrieving additional
+	// graph data.
+	GitHubRepo string `json:"gitHubRepo"`
+
+	// Branch is the git branch to use on GitHub for retrieving additional graph
+	// data.
+	Branch string `json:"branch"`
+}
+
+// CincinnatiStatus defines the observed state of Cincinnati
+type CincinnatiStatus struct {
+	// Conditions describe the state of the Cincinnati resource.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []conditionsv1.Condition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// Condition Types
+const (
+	// ConditionResourcesExist reports whether all required resources have been created
+	// in the cluster and reflect the specified state.
+	ConditionResourcesExist conditionsv1.ConditionType = "ResourcesExist"
```

Comment:
I dislike the name of this, but I wasn't feeling creative enough to come up with something better. Suggestions are encouraged. This condition really just means that reconciliation completed without error.

Reply:
How about `ConditionClusterResourcesReady` instead of `ConditionResourcesExist`?

Reply:
> How about `ConditionClusterResourcesReady` instead of `ConditionResourcesExist`?

This condition only signifies that the operator did all of its work. For example, this operator ensured that a Deployment exists. But, it does not guarantee that the Deployment itself is "ready".

Reply:
Yeah I used [`ReconcileCompleted`](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/f8f8f2c6ba1b0eab5e745faf917b1351425b5231/pkg/controller/hyperconverged/hyperconverged_controller.go#L61) to signify that we made it through and did all we wanted to. Not sure if it is what you want.